actors/miner/src/testing.rs

```diff
@@ -75,17 +74,14 @@ pub fn check_state_invariants<BS: Blockstore>(
                         "on chain sector's sector number has not been allocated {sector_number}"
                     ),
                 );
-                sector.deal_ids.iter().for_each(|&deal| {
-                    miner_summary.deals.insert(
-                        deal,
+                if !sector.deal_weight.is_zero() || !sector.verified_deal_weight.is_zero() {
+                    miner_summary.live_data_sectors.insert(
+                        sector_number,
                         DealSummary {
                             sector_start: sector.activation,
                             sector_expiration: sector.expiration,
```

Comment:
Suggestion: include the deal weights so we can verify against deals and allocations separately, since either or both might be in use.